ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2.java

```diff
@@ -63,23 +63,13 @@
       final StateQueryResult<ValueAndTimestamp<GenericRow>>
           result = stateStore.getKafkaStreams().query(request);
 
-      final Map<Integer, QueryResult<ValueAndTimestamp<GenericRow>>> queryResultMap =
-          result.getPartitionResults();
-      if (queryResultMap.size() != 1) {
-        throw new MaterializationException("The query did not return exactly one partition result: "
-                + result.getPartitionResults());
-      }
-      if (!queryResultMap.containsKey(partition)) {
-        throw new MaterializationException(
-            "The query did not return the requested partition result: "
-                + result.getPartitionResults());
-      }
       final QueryResult<ValueAndTimestamp<GenericRow>> queryResult =
-          queryResultMap.get(partition);
+          result.getPartitionResults().get(partition);
+      // Some of these failures are retriable, and in the future, we may want to retry
+      // locally before throwing.
       if (queryResult.isFailure()) {
-        throw new MaterializationException(queryResult.getFailureMessage());
-      }
-      if (queryResult.getResult() == null) {
```

Comment:
`if {throw/return} else if ...` is generally preferable to `if {throw/return} if ...` even through they're functionally equivalent because it makes the control flow clearer.